src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.Html.Core.Tree;
+using Microsoft.Html.Core.Tree.Nodes;
+using Microsoft.R.Components.PackageManager.Model;
+using Microsoft.R.Components.Test.PackageManager;
+
+namespace Microsoft.R.Components.PackageManager.Implementation {
+    internal class RPackageWebParser {
+        /// <summary>
+        /// Downloads the index html page specified, and extracts a set of known fields from it.
+        /// </summary>
+        /// <param name="packageUri">Location of index.html page.</param>
+        /// <param name="result">RPackage object to fill in. A new one is created if none is passed in.</param>
+        /// <returns>RPackage object populated with the information found in the html page.</returns>
+        /// <remarks>
+        /// If an existing RPackage object is passed in, the fields that are already filled in
+        /// won't be overwritten by the ones extracted from the html page.
+        /// </remarks>
+        public static async Task<RPackage> RetrievePackageInfo(Uri packageUri, RPackage result = null) {
+            string content;
+            try {
+                using (var webClient = new WebClient()) {
+                    content = await webClient.DownloadStringTaskAsync(packageUri);
+                }
+            } catch (WebException ex) {
+                throw new RPackageInfoRetrievalException(string.Format("Error reading index page from {0}", packageUri), ex);
```

Comment:
Intent was that the caller of RPackageManager, so yes probably view model.  The alternative was that RPackageManager catches it, but then view model wouldn't know things went wrong and would have no way of indicating that to the user.